Working_script.R

```diff
@@ -0,0 +1,195 @@
+# Libraries required
+#install.packages("devtools")
+#devtools::install_github("ropensci/FedData")
+library(readr)
+library(tidyverse)
+library(FedData)
+library(raster)
+library(sf)
+library(spData)
+library(leaflet)
+library(purrr)
+# Data gathering
+tdir=tempdir()
+## Tick data
+adulturl <- "https://health.data.ny.gov/api/views/vzbp-i2d4/rows.csv?accessType=DOWNLOAD&bom=true&query=select+*"
+nymphurl <- "https://health.data.ny.gov/api/views/kibp-u2ip/rows.csv?accessType=DOWNLOAD&bom=true&query=select+*"
+if(file.exists(paste(tdir,"/Adultdata.csv",sep=""))==FALSE){
+download.file(adulturl, destfile = file.path(tdir,"/Adultdata.csv"))}
+if(file.exists(paste(tdir,"/Nymphdata.csv",sep=""))==FALSE){
+download.file(nymphurl,destfile = file.path(tdir,"/Nymphdata.csv"))}
+Nymphdata <- read_csv(paste(tdir,"/Nymphdata.csv",sep=""))
+Adultdata <- read_csv(paste(tdir,"/Adultdata.csv",sep=""))
+
+#NYS Shapefiles
+shpurl="http://gis.ny.gov/gisdata/fileserver/?DSID=927&file=NYS_Civil_Boundaries.shp.zip"
+if(file.exists(paste(tdir,"/NYS_Civil_Boundaries_SHP/Counties_Shoreline.shp",sep=""))==FALSE){
+download.file(shpurl, destfile = file.path(tdir,"Boundaries.zip"))
+unzip(file.path(tdir,"Boundaries.zip"),exdir = tdir)
+Cnt=read_sf(paste(tdir,"/NYS_Civil_Boundaries_SHP/Counties_Shoreline.shp",sep=""))}
+#list.files(paste(tdir,"/NYS_Civil_Boundaries_SHP",sep=""))
+County_Boundaries = Cnt[,c(1,18)]
+NYS = st_union(County_Boundaries)
+simpSpNYS=st_simplify(NYS,
+                      dTolerance=100)
+SpNYS=as(NYS,Class="Spatial")
+
+
+
+
+## Land use data
+
+data(us_states)
+NY = us_states[17,]
+NLCD=get_nlcd(template = st_as_sf(NY),
+              label = "NYS",
+              year = 2016,
+              dataset='Land_Cover',
+              landmass="L48")
+
+# Start plotting
```

Comment:
Ugh - ugly formatted data.  Instead of the next few lines, I suggest renaming the columns in one command - e.g. `colnames(Adultdata)=c()` and then trying to find a way to 'gather' the columns.  I think this would be more concise and elegant.  Lots of chances to make a mistake doing it manually like this (e.g. getting a column ID wrong).

Reply:
Agreed. I believe I started this before learning that 'gather' method. If I were doing this again, I would start with that rather than all of this manual line by line tidying.